hw/top_earlgrey/data/pins_cw310_hyperdebug.xdc

```diff
@@ -0,0 +1,147 @@
+set_property -dict { PACKAGE_PIN A9  IOSTANDARD LVCMOS33 }  [get_ports { IOC1  }]; # J5.15 USERIOB-15
+set_property -dict { PACKAGE_PIN D8  IOSTANDARD LVCMOS33 }  [get_ports { IOC2  }]; # J5.16 USERIOB-16
+set_property -dict { PACKAGE_PIN G15 IOSTANDARD LVCMOS33 }  [get_ports { IOC3  }]; # J4.12 USERIOA-12
+set_property -dict { PACKAGE_PIN F15 IOSTANDARD LVCMOS33 }  [get_ports { IOC4  }]; # J4.14 USERIOA-14
+set_property -dict { PACKAGE_PIN D15 IOSTANDARD LVCMOS33 }  [get_ports { IOC5  }]; # J4.16 USERIOA-16
+set_property -dict { PACKAGE_PIN D16 IOSTANDARD LVCMOS33 }  [get_ports { IOC6  }]; # J4.18 USERIOA-18
+set_property -dict { PACKAGE_PIN P18 IOSTANDARD LVCMOS33 }  [get_ports { IOC7  }]; # USRUSB_VBUS_DETECT
+set_property -dict { PACKAGE_PIN D9  IOSTANDARD LVCMOS33 }  [get_ports { IOC8  }]; # J5.18 USERIOB-18
+set_property -dict { PACKAGE_PIN N19 IOSTANDARD LVCMOS33 }  [get_ports { IOC9  }]; # PMOD2_IO1 (GPIO)
+set_property -dict { PACKAGE_PIN F19 IOSTANDARD LVCMOS33 }  [get_ports { IOC10 }]; # J4.36 USERIOA-36
+set_property -dict { PACKAGE_PIN G20 IOSTANDARD LVCMOS33 }  [get_ports { IOC11 }]; # J4.38 USERIOA-38
+set_property -dict { PACKAGE_PIN B19 IOSTANDARD LVCMOS33 }  [get_ports { IOC12 }]; # J4.40 USERIOA-40
+
+
+
+## IOR bank
+set_property -dict { PACKAGE_PIN  E20  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR0  }]; # J4.24 USERIOA-24
+set_property -dict { PACKAGE_PIN  C18  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR1  }]; # J4.26 USERIOA-26
+set_property -dict { PACKAGE_PIN  D20  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR2  }]; # J4.28 USERIOA-28
+set_property -dict { PACKAGE_PIN  F20  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR3  }]; # J4.21 USERIOA-21
+set_property -dict { PACKAGE_PIN  D19  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR4  }]; # J4.23 USERIOA-23
+set_property -dict { PACKAGE_PIN  C19  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR5  }]; # J4.25 USERIOA-25
+set_property -dict { PACKAGE_PIN  K17  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR6  }]; # J4.27 USERIOA-27
+set_property -dict { PACKAGE_PIN  K18  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR7  }]; # J4.29 USERIOA-29
+set_property -dict { PACKAGE_PIN  G17  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR8  }]; # J4.31 USERIOA-31
+set_property -dict { PACKAGE_PIN  H18  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR9  }]; # J4.33 USERIOA-33
+set_property -dict { PACKAGE_PIN  F17  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR10 }]; # J4.35 USERIOA-35
+set_property -dict { PACKAGE_PIN  G19  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR11 }]; # J4.30 USERIOA-30
+set_property -dict { PACKAGE_PIN  F18  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR12 }]; # J4.37 USERIOA-37
+set_property -dict { PACKAGE_PIN  E15  IOSTANDARD LVCMOS33 PULLTYPE PULLUP } [get_ports { IOR13 }]; # J4.39 USERIOA-39
```

Comment:
Oops, setting these all to PULLUP was a copy-paste accident. We probably want to have any pull-ups come from outside, right? (whether from the hyperdebug's weak pull-up or an actual resistor on an adapter board)

In addition, IOR0-IOR4 used the internal pull-ups because they were dedicated to JTAG on the SAM3X variant, but maybe those should be removed for the case of connecting to hyperdebug, since their function can be software-controlled.

Reply:
I went ahead and removed the pull-ups.